app/javascript/src/lib/codeActionProviders/transformParameterObjectsIntoPositionalParameters.js

```diff
@@ -0,0 +1,110 @@
+import { completionsMap, settings } from "../../stores/editor"
+import { getClosingBracket, getPhraseFromPosition, splitArgumentsString } from "../../utils/parse"
+import { get } from "svelte/store"
+import { parseParameterObjectContent } from "../../utils/compiler/parameterObjects"
+import { getIndentForLine } from "../../utils/codemirror/indent"
+
+/**
+ * @type {import("../codeActions").CodeActionProvider}
+ */
+export function transformParameterObjectsIntoPositionalParameters({ view, state }) {
+  const actions = []
+
+  const { from: cursorFrom, to: cursorTo } = state.selection.main
+  if (cursorFrom !== cursorTo) return
+
+  const line = state.doc.lineAt(cursorFrom)
+  if (!line.text) return null
+
+  const phrase = getPhraseFromPosition(line, cursorFrom)
+  const completion = get(completionsMap).find(({ type, label }) => type === "function" && label === phrase.text)
+  if (!completion) return
+
+  const fileWideActionNameStart = line.from + phrase.start
+  const fileWideActionNameEnd = line.from + phrase.end + 1
+
+  const actionCallCont = state.doc.sliceString(fileWideActionNameEnd, fileWideActionNameEnd + 100)
+  const openParenIndex = actionCallCont.match(/(?<=^\s*)\(/)?.index ?? -1
+  if (openParenIndex < 0) return
+
+  const closingParenIndex = getClosingBracket(" " + actionCallCont, "(", ")", 0) // this function is cursed...
+  if (closingParenIndex < 0) return
+
+  const fileWideParenFrom = fileWideActionNameEnd + openParenIndex
+  const fileWideParenTo = fileWideActionNameEnd + closingParenIndex
+
+  // empty positional parameters
+  if (fileWideParenTo - fileWideParenFrom <= "()".length) return
+
+  const openParameterObjectBracketIndex = actionCallCont.slice(openParenIndex + 1).match(/(?<=^\s*)\{/)?.index ?? -1
+  if (openParameterObjectBracketIndex >= 0) {
+    // parameter object → positional parameters
+
+    const closingParameterObjectBracketIndex = getClosingBracket(actionCallCont, "{", "}", 0)
+    if (closingParameterObjectBracketIndex < 0) return
+
+    const fileWideParameterObjectFrom = (fileWideParenFrom + 1) + openParameterObjectBracketIndex
+    const fileWideParameterObjectTo = (fileWideParenFrom + 1) + closingParameterObjectBracketIndex
+
+    actions.push({
+      label: "Transform object into positional parameters",
```

Comment:
Perhaps we can change the wording to "regular parameters" or "workshop syntax" as "positional parameters" is a bit technical

Reply:
I was hoping people would learn what positional parameters are along the way, but sure.

"Positional parameters" sounds weird anyway, since normally these would be called "positional arguments"[[1]]. But we already called the object notation "parameter objects" instead of "argument objects", so eh.

[1]: https://stackoverflow.com/a/156787